File: CPP/Program/Clothing.cpp
/*Saya Muhammad Rifky Afandi dengan NIM 2202346 mengerjakan LP2 dalam mata kuliah 
Desain Pemrograman Berbasis Objek untuk keberkahanNya maka saya tidak 
melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/


//import library
#include <iostream>
#include <string>
#include "Product.cpp" //import file product 

//menggunakan namespace standar
using namespace std;

// Inheritance dengan file product
class Clothing : public Product{

    // Atribut privat
    private:
        string size;
        string material;
        string gender;
    
    // Atribut Publik
    public:
        // Constructor
        Clothing(){
            this->size = "";
            this->material = "";
            this->gender = "";
        }
        // Constructor dengan parameter
        Clothing(string size, string material, string gender){
            this->size = size;
            this->material = material;
            this->gender = gender;
        }

        // Getter dan Setter
        string getSize(){
            return this->size;
        }

        void setSize(string size){
            this->size = size;

        }

        string getMaterial(){
            return this->material;
        }

        void setMaterial(string material){
            this->material = material;
        }

        string getGender(){
            return this->gender;
        }

        void setGender(string gender){
            this->gender = gender;
        }

        ~Clothing(){

        }

};

File: Python/Program/Clothing.py
## Saya Muhammad Rifky Afandi dengan NIM 2202346 mengerjakan LP2 dalam mata kuliah 
## Desain Pemorograman Berbasis Objek untuk keberkahanNya maka saya tidak 
## melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

## import file
from Product import Product

## inheritance dengan file Product
class Clothing(Product):
    ## Atribut privat
    __size = ""
    __material = ""
    __gender = ""

    ## Constructor
    ## Atribut Publik
    def __init__(self, size = "", material = "", gender = ""):
        self.__size = size
        self.__material = material
        self.__gender = gender
    
    ## Getter dan Setter    
        
    def get_size(self):
        return self.__size
    
    def set_size(self, size):
        self.__size = size
        
    def get_material(self):
        return self.__material
    
    def set_material(self, material):
        self.__material = material
        
    def get_gender(self):
        return self.__gender
    
    def set_gender(self, gender):
        self.__gender = gender
    
    
    